kolla/tests/test_build.py

```diff
@@ -57,15 +57,26 @@ def setUp(self):
         self.imageChild.parent = self.image
         self.imageChild.path = self.useFixture(fixtures.TempDir()).path
 
+    @mock.patch('docker.version', '2.7.0')
     @mock.patch.dict(os.environ, clear=True)
     @mock.patch('docker.APIClient')
-    def test_push_image(self, mock_client):
+    def test_push_image_before_v3_0_0(self, mock_client):
         self.dc = mock_client
         pusher = build.PushTask(self.conf, self.image)
         pusher.run()
         mock_client().push.assert_called_once_with(
             self.image.canonical_name, stream=True, insecure_registry=True)
 
+    @mock.patch('docker.version', '3.0.0')
```

Comment:
Do these patches upgrade/downgrade docker version in the host machine?

Reply:
Almost - the mock simulates that version 3.0.0 is installed for the purposes of the unit test. It doesn't actually matter what is on the host. If you haven't seen mocks before they're definitely worth checking out. They can make testing much easier / more effective.

Reply:
Now I see! Please ignore my question about handling the range.